endpoint.go

```diff
@@ -637,6 +639,14 @@ func EndpointOptionGeneric(generic map[string]interface{}) EndpointOption {
 	}
 }
 
+// CreateOptionIpam function returns an option setter for the ipam configuration for this endpoint
+func CreateOptionIpam(prefAddress net.IP, ipamOptions map[string]string) EndpointOption {
```

Comment:
@mavenugo My advise is to support the full extend of networking otherwise the architecture is not future proof. Is there a specific reason to add these restrictions and limit the usability of docker ? Are you saying that you do not envision docker being used in the following use cases:
*   Anycast DNS
*   Load Balancer Direct Server Return
*   Cloudflare like anycast in DC (https://blog.cloudflare.com/cloudflares-architecture-eliminating-single-p/)

I fail to see the benefit of limiting the number of IP configured per container, mandate uniqueness of IP, or subnet membership since they are perfectly legal networking options, and more over when we are implementing both a custom Network driver and a custom IPAM driver.